kolibri/core/device/serializers.py

```diff
@@ -266,6 +269,33 @@ class Meta:
     def create(self, validated_data):
         raise serializers.ValidationError("Device settings can only be updated")
 
+    def update(self, instance, validated_data):
+        if "extra_settings" in validated_data:
+            updated_extra_settings = validated_data.get("extra_settings")
+            initial_extra_settings = getattr(instance, "extra_settings", "{}")
+
+            if updated_extra_settings != initial_extra_settings:
+                automatic_download_enabled = updated_extra_settings.get(
+                    "enable_automatic_download"
+                )
+                if (
+                    automatic_download_enabled
+                    and automatic_download_enabled
+                    != initial_extra_settings.get("enable_automatic_download")
+                ):
+                    # Cancel any ongoing syncing before we kick start the new one
+                    job_storage.cancel_if_exists(SYNC_CANCEL_STATIC_ID)
```

Comment:
Hrm - noting that this will cancel it even if it is currently running - which doesn't seem super ideal. I think we can merge this for now though, and resolve this in https://github.com/learningequality/kolibri/issues/11007 instead.